src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4486,40 +4472,24 @@ void makeAndAdjustReceiverSlot(BoundExpression receiver)
                 return null;
             }
 
-            BoundExpression consequence;
-            BoundExpression alternative;
-            Conversion consequenceConversion;
-            Conversion alternativeConversion;
-            bool consequenceEndReachable;
-            bool alternativeEndReachable;
+            (var consequence, var consequenceConversion, consequenceRValue) = visitConditionalOperand(consequenceState, originalConsequence);
+            var consequenceConditionalState = PossiblyConditionalState.Create(this);
+            Unsplit();
+            consequenceState = this.State.Clone();
+            var consequenceEndReachable = consequenceState.Reachable;
 
-            // In cases where one branch is unreachable, we don't need to Unsplit the state
-            if (!alternativeState.Reachable)
-            {
-                (alternative, alternativeConversion, alternativeRValue) = visitConditionalOperand(alternativeState, originalAlternative);
-                (consequence, consequenceConversion, consequenceRValue) = visitConditionalOperand(consequenceState, originalConsequence);
-                alternativeEndReachable = false;
-                consequenceEndReachable = IsReachable();
-            }
-            else if (!consequenceState.Reachable)
+            (var alternative, var alternativeConversion, alternativeRValue) = visitConditionalOperand(alternativeState, originalAlternative);
```

Comment:
From our discussion, it may be possible to turn this into:
```
var alternativeConditionalState = PossiblyConditionalState.Create(this);
alternativeState = alternativeConditionalState.CloneAndUnsplit();
Join(ref consequenceConditionalState);
```

Then we could consider similar code for the consequence above:
```
var consequenceConditionalState = PossiblyConditionalState.Create(this);
consequenceState = consequenceConditionalState.CloneAndUnsplit();
```